src/main/java/com/netflix/imflibrary/writerTools/utils/SequenceDurationValidator.java

```diff
@@ -0,0 +1,67 @@
+package com.netflix.imflibrary.writerTools.utils;
+
+import com.netflix.imflibrary.st2067_2.IMFTrackFileResourceType;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class SequenceDurationValidator {
+
+    /**
+     * Validates and adjusts the duration of the resources in the sequence to ensure the total duration
+     * is an integer number of Composition Edit Units.
+     *
+     * @param resources the list of resources in the sequence
+     * @param compositionEditRateNumerator of the composition edit rate
+     * @param compositionEditRateDenominator of the composition edit rate
+     * @return a new list of resources with adjusted durations if needed
+     */
+    public static List<IMFTrackFileResourceType> validateAndAdjustResourceDurations(List<IMFTrackFileResourceType> resources, long compositionEditRateNumerator, long compositionEditRateDenominator) {
+        long totalResourceDuration = resources.stream()
+                .mapToLong(resource -> {
+                    if (resource != null) {
+                        return resource.getSourceDuration().longValue();
+                    }
+                    return 0;
+                })
+                .sum();
+
+        List<IMFTrackFileResourceType> adjustedResources = new ArrayList<>(resources);
+        IMFTrackFileResourceType lastResource = adjustedResources.get(adjustedResources.size() - 1);
+        long resourceEditRateNumerator = lastResource.getEditRate().getNumerator();
+        long resourceEditRateDenominator = lastResource.getEditRate().getDenominator();
+        long remainder = 0;
+
+        if (resourceEditRateDenominator == 1001) {
+            // Non-integer frame rates must be evenly divisible by 5 in order to align with audio samples
```

Comment:
Unfortunately, the math gets more complicated than this.

The issue isn't _entirely_ related to non-integer rates. But instead, it is caused when the rates of two tracks are not multiples with one another, such that each video frame ends _exactly_ on the bounds of an audio sample.

That means that both the video AND audio rate are relevant. Luckily, the IMF core constraints limit the audio sample rate to only a few rates that are easy to work with (`2067-2:2020  5.3.2.2`). So really, we _only_ need to worry about the video frame rate.

Most of the common non-integer video frame rates _actually are_ compatible with the allowed audio sample rates.  `24000/1001 fps` to `48kHz` works out to be _exactly_ `2002` samples-per-frame.
The only common frame rates that are _really_ problematic are `30000/1001` and `60000/1001`. Thats where you end up needing to align on every fifth video frame.

But this check should still allow non-integer frame rates for `23.976` and `47.952`